ERCS/erc-7941.md

```diff
@@ -0,0 +1,573 @@
+
+## Specification
+This section outlines the technical specifications for implementing the Smart Creative Commons Zero (SCC0) license for public decentralized applications (dApps/dAIpps). It defines a standardized on-chain framework that allows smart contracts to declare SCC0 compliance, supports automated verification, and facilitates decentralized governance.
+
+**1. Compliance Contracts of v1 and v2**
+
+**SCC0 v1 Compliance Contract**
+
+SCC0 v1 has been deployed by DAism, and any dApp/dAIpp adhering to it must:
+
+- Interact with DAism's smart contract `0xdFBF69B7E5366FB3001C9a214dd85c5FE3f90bAe`. Or go to DAism to mint a smart common.
+
+- SCC0 v1 Compliance Contract which is deployed by DAism:
+
+```solidity
+contract SCC0License {
+    string public constant LICENSE_NAME = "SCC0";
+    uint8 public constant VERSION = 1;
+    bool public constant REJECT_PRIVATE_APP = true;
+    bool public constant SELF_ISSUED_TOKEN = false;
+    bool public constant ANONYMITY_ENSURED = true;
+    bool public constant NO_RIGHTS = true;
+    bool public constant NO_LIABILITY = true;
+    bool public constant CODE_AUDITED_VERIFIED_PUBLISHED = true;
+    bool public constant PERMANENTLY_FREE = true;
+    address public constant PUBLIC_GOVERNANCE_FUND = 0xe40b05570d2760102c59bf4ffc9b47f921b67a1F;
+}
+```
+
+- DAism has defined the Smart Common structure:
+
+```solidity
+struct SCInfo {
+    string name;        // Name of the smart common
+    string symbol;      // Symbol of the smart common
+    string desc;        // Description of the smart common
+    address manager;    // Address of the smart common manager
+    uint16 version;     // Version number of the smart common
+    string SCType;      // Type of the smart common
+}
+```
+
+- Additional mappings and governance structures are included for community interactions:
+
+```solidity
+mapping(address => Object.Member) public memberInfos; // Stores Smart Common members and their dividend ratios
+uint32 public proposalLifetime; // Validity period of Smart Common proposals
+uint32 public proposalCoolingPeriod; // Cooling period for Smart Common proposals
+uint16 public strategy; // Pass rate for Smart Common proposals
+mapping(uint => File) public logoStorages; // Storage for Smart Common logos
+```
+- Verify if it is scc0 v1 version using the following methods:
+```solidity
+interface IDaism {
+    //check the address whether daism sc type of dapp
+    function dappToSC(address dApp) external view returns (uint);
+}
+
+// Check if a dApp is idaism dapp(scc0 v1 version)
+//daismAddress=0xdFBF69B7E5366FB3001C9a214dd85c5FE3f90bAe
+function isDaismSC(address _dApp) external view returns (bool) {
+     return IDaism(daismAddress).dappToSC(_dApp)>0;
+}
+```
+
```

Comment:
So far this looks like it's documenting a particular deployed contract, and not a standard that requires multiple parties to coordinate. If you expect everyone to interact with _this_ contract, then you should probably go write some developer documentation. You don't need an ERC.

ERCs are only really useful when multiple parties need to interoperate without having to communicate. ERC-20 tokens are a great example of a good standard. Every wallet wants to display every ERC-20 token, while every ERC-20 token wants to work with every wallet. If these parties had to write bespoke code for every integration, the whole ecosystem wouldn't have gone anywhere. That's where ERCs shine. Here, it looks like you have a single contract. If that's true, you should just publish some docs somewhere.